elastic/kibana
# [Ingest Pipelines] - Fix enrich processor override field

Fixes https://github.com/elastic/kibana/issues/148105

## Summary

Small fix for the enrich processor which was not correctly serializing the override field. Also added a few tests which were missing.

**How to test**
1. Start elasticsearch with `yarn es snapshot --license=trial` and kibana with `yarn start`
2. Navigate to `Stack Management` -> `Ingest Pipelines` and create a new one
3. Add a new enrich processor to the newly created pipeline and
4. Verify that:
    * When leaving the override as true, the `show request` doesnt show the override in the request (since its true by default)
    * Edit the processor and set the override as false, and check that in the `show request` the override is set to false
    * Check that saving the pipeline also persists the state correctly (you might have to create a test enrich policy first)
